### PR TITLE
Add New Cryptocurrency

### PR DESCRIPTION
**Pull request by FedeRosaless**

### CODE REVIEW

File: src/screens/AddCrypto/style.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  border: 2px solid ${({theme}) => theme.colors.gray};
+  width: 90%;
+  height: 60px;
+  position: relative;
+  top: 29%;
+  margin: 5%;
+  border-radius: 6px;
+  padding-left: 10px;
+  font-size: 17px;
+  background-color: #f1f1f154;
```
Comment (JustBecerra) [inline src/screens/AddCrypto/style.ts]: you should use theme colors instead of inline colors

File: src/screens/AddCrypto/index.tsx [hunk 2]
```diff
@@ -1,10 +1,71 @@
...
+  const [borderColor, setBorderColor] = useState(theme.colors.borderGray);
+  const {navigate} = useNavigation();
+
+  const cryptos = useSelector(
+    (state: RootState) => state.selectedCrypto.cryptos,
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  const findDuplicate = () => {
+    const res = cryptos.find(e => {
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: Use implicit return

File: src/screens/AddCrypto/index.tsx [hunk 3]
```diff
@@ -1,10 +1,71 @@
...
+      return (
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx:22]: Use implicit return

File: src/screens/AddCrypto/index.tsx [hunk 4]
```diff
@@ -1,10 +1,71 @@
...
+        e.name.toLowerCase() === text.toLowerCase() ||
+        e.symbol.toLowerCase() === text.toLowerCase()
+      );
+    });
+    return res;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: Use implicit return

File: src/screens/CryptoList/index.tsx [hunk 5]
```diff
@@ -4,22 +4,15 @@ import CryptoItem from '../../components/CryptoItem';
...
-import {getCrypto} from '../../store/actions';
-import {AppDispatch, RootState} from '../../store/index';
+import {useSelector} from 'react-redux';
+import {RootState} from '../../store/index';
 
 const CryptoList = () => {
   const {navigate} = useNavigation();
-  const dispatch = useDispatch<AppDispatch>();
   const crypto = useSelector(
     ({selectedCrypto}: RootState) => selectedCrypto.cryptos,
```
Comment (FedericoLeiva12) [inline src/screens/CryptoList/index.tsx]: Destructuring: ({selectedCrypto: {cryptos}})

File: src/screens/AddCrypto/style.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+`;
+
+export const AddText = styled.Text`
+  font-size: 26px;
+  color: ${({theme}) => theme.colors.darkGray};
+  position: relative;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: You should avoid the use of position using flex instead

File: src/screens/AddCrypto/style.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+export const Container = styled.View`
+  display: flex;
+`;
+
+export const CryptoInput = styled.TextInput`
+  border: 2px solid ${({theme}) => theme.colors.gray};
+  width: 90%;
+  height: 60px;
+  position: relative;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: You should avoid the use of position using flex instead

File: src/screens/AddCrypto/index.tsx [hunk 8]
```diff
@@ -1,10 +1,71 @@
...
+        onChangeText('');
+        navigate('CryptoList');
+      }
+    } catch (err) {
+      return err;
+    }
+  };
+  return (
+    <Modal animationType="fade">
+      <TouchableOpacity onPress={() => navigate('CryptoList' as never)}>
```
Comment (tomasdematos) [inline src/screens/AddCrypto/index.tsx]: as never? 
This is not a correct type, check the type configuration of your navigation

File: src/screens/AddCrypto/index.tsx [hunk 9]
```diff
@@ -1,10 +1,71 @@
...
+import {useState} from 'react';
+import {CryptoInput, AddText, BackText, ModalText, Touchable} from './style';
+import {useDispatch, useSelector} from 'react-redux';
+import {AppDispatch, RootState} from '../../store';
+import {useNavigation} from '@react-navigation/native';
+import {getCrypto} from '../../store/actions';
+import theme from '../../theme';
+
+const AddCrypto = () => {
+  const [text, onChangeText] = useState('');
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: The name of "onChangeText" should be "setText"

File: src/screens/AddCrypto/style.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  font-family: Inter;
+`;
+
+export const ModalText = styled.Text`
+  font-size: 18px;
+  color: ${({theme}) => theme.colors.gray};
+  background-color: ${({theme}) => theme.colors.yellow};
+  text-align: center;
+  text-align-vertical: center;
+  width: 160px;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: You should avoid setting width and height, instead you can use margin/padding and set the distance between the element and the border of the screen / parent component

File: src/screens/AddCrypto/style.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+  height: 48px;
+  border-radius: 6px;
+`;
+
+export const Touchable = styled.TouchableOpacity`
+  width: 155px;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: You should avoid setting width and height, instead you can use margin/padding and set the distance between the element and the border of the screen / parent component

File: src/screens/AddCrypto/style.ts [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  left: 6%;
+  line-height: 30px;
+  font-weight: 700;
+  font-family: Inter;
+  letter-spacing: 0px;
+`;
+
+export const BackText = styled.Text`
+  font-Size: 17px;
+  line-Height: 24px;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: line-height*

File: src/screens/AddCrypto/style.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+  font-Weight: 400;
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/style.ts]: font-weight*

File: src/screens/AddCrypto/index.tsx [hunk 14]
```diff
@@ -1,10 +1,71 @@
...
+  const [borderColor, setBorderColor] = useState(theme.colors.borderGray);
+  const {navigate} = useNavigation();
+
+  const cryptos = useSelector(
+    (state: RootState) => state.selectedCrypto.cryptos,
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: Use destructuring

File: src/screens/AddCrypto/index.tsx [hunk 15]
```diff
@@ -1,10 +1,71 @@
...
+      <AddText>Add a Cryptocurrency</AddText>
+      <CryptoInput
+        onBlur={() => setBorderColor(theme.colors.borderGray)}
+        onFocus={() => setBorderColor(theme.colors.yellow)}
+        style={{borderColor}}
+        onChangeText={onChangeText}
+        placeholder="Use a name or ticker symbol..."
+      />
+      <Touchable onPress={handleCrypto}>
+        <ModalText
```
Comment (tomasdematos) [inline src/screens/AddCrypto/index.tsx]: The button should have the prop, disabled, if the text is empty, so the validation in the line 32 is unnecessary,
also don't use inline style, validate the color in the styled

File: src/screens/AddCrypto/index.tsx [hunk 16]
```diff
@@ -1,10 +1,71 @@
...
+  return (
+    <Modal animationType="fade">
+      <TouchableOpacity onPress={() => navigate('CryptoList' as never)}>
+        <BackText>&lt; Back to list</BackText>
+      </TouchableOpacity>
+      <AddText>Add a Cryptocurrency</AddText>
+      <CryptoInput
+        onBlur={() => setBorderColor(theme.colors.borderGray)}
+        onFocus={() => setBorderColor(theme.colors.yellow)}
+        style={{borderColor}}
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: you should use styled components

File: src/screens/AddCrypto/index.tsx (hunk 15 above)
Comment (tomasdematos) [reply src/screens/AddCrypto/index.tsx]: also why it's name is ModalText? it should be ButtonText

File: src/screens/AddCrypto/index.tsx [hunk 17]
```diff
@@ -1,10 +1,73 @@
...
+
+  const handleFocusInput = () => {
+    setFocused(true);
+  };
+
+  const {
+    selectedCrypto: {cryptos},
+  } = useSelector((state: RootState) => state);
+
+  // const crypto = useSelector(
```
Comment (FedericoLeiva12) [inline src/screens/AddCrypto/index.tsx]: delete these comments

Comment (VicenBayugar) [inline src/screens/AddCrypto/index.tsx]: delete this comment

File: src/screens/CryptoList/index.tsx [hunk 18]
```diff
@@ -4,28 +4,25 @@ import CryptoItem from '../../components/CryptoItem';
...
-    ({selectedCrypto}: RootState) => selectedCrypto.cryptos,
-  );
+  const {
+    selectedCrypto: {cryptos},
+  } = useSelector((state: RootState) => state);
 
-  useEffect(() => {
-    dispatch(getCrypto('bitcoin'));
-  }, []);
+  // const crypto = useSelector(
```
Comment (VicenBayugar) [inline src/screens/CryptoList/index.tsx]: Delete this comment

**Pull request merged**